src/main/java/org/folio/dew/batch/bulkedit/jobs/BulkEditHoldingsProcessor.java

```diff
@@ -144,13 +153,27 @@ private ExtendedHoldingsRecordCollection getHoldingsRecords(ItemIdentifier itemI
       }
     } else {
       // Process local tenant case
+      checkReadPermissions(folioExecutionContext.getTenantId(), identifier);
       var holdingsRecordCollection = getHoldingsRecordCollection(type, itemIdentifier);
       return new ExtendedHoldingsRecordCollection().extendedHoldingsRecords(holdingsRecordCollection.getHoldingsRecords().stream()
           .map(holdingsRecord -> new ExtendedHoldingsRecord().tenantId(folioExecutionContext.getTenantId()).entity(holdingsRecord)).toList())
         .totalRecords(holdingsRecordCollection.getTotalRecords());
     }
   }
 
+  private void checkReadPermissions(String tenantId) {
```

Comment:
Both these methods looks close or even the same like for items. Is it possible to move all of them and possibly unify from processors to one component or to permissions validator?

Reply:
Hi @khandramai , one of the methods was moved to the validator logic. The other one requires specific data for certain records processors.